posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,506 @@
+
+The transformer may now be run using the default rules:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location>
+--
+
+=== Running the transformer with custom rules
+
+The transformer may be run using custom rules.  This is done by specifying additional arguments on the command line.  The additional arguments specify local rules files that the transformer is to be used instead of the default rules files.
+
+For example, Open-Liberty has more up to date rules for performing transformations.  To use the Open-Liberty rules, download these from the Open-Liberty GIT repository, from folder:
+
+    https://github.com/OpenLiberty/open-liberty/tree/integration/dev/wlp-jakartaee-transform/rules
+
+Then add three command line arguments to the transformer command line invocation:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location> \
+    -tr ${rules}/jakarta-renames.properties \
+    -td ${rules}/jakarta-direct.properties \
+    -tf ${rules}/jakarta-xml-master.properties
+--
+
+Usually, just three additional command line arguments are needed.  There are additional command line arguments for transforming bundle archives, and for more finely selecting which target files are to be transformed, but those additional arguments are usually not needed.
+
+The target location may be a packaged application archive -- EAR, WAR, RAR, or JAR files -- or may be a directory containing the fully or partially expanded contents of one of these archives.  The target can also be a directory containing java source, or a directory containing several application archives.  All of the artifacts in the target location are processed by the transformer.
+
+=== More about transforming server configurations
+
+Transformation of a server configuration is only necessary when java EE package names are exposed in the server configuration.  One case of explosed java EE packages names are JMS activation specification elements.  For example:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="javax.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+This must be transformed to:
+
+[source]
+--
+  <jmsActivationSpec id="SharedSubscriptionWithMsgSel/TestTopic1">
+    <properties.wasJms
+        destinationRef="jms/FAT_TOPIC"
+        destinationType="jakarta.jms.Topic"
+        subscriptionDurability="DurableShared"
+        clientId="cid1"
+        subscriptionName="DURSUB"/>
+  </jmsActivationSpec>
+--
+
+== Updating features in server configurations:
+
+When running a Jakarta EE 9 enable application with Open-Liberty, Jakarta EE 9 versions of Open-Liberty features must be specified in the server configuration.
```

Comment:
I would combined these two sentences into the same paragraph as:

`When running a Jakarta EE 9 application with Open Liberty, Jakarta EE 9 versions of Open Liberty features must be specified in the server configuration. In some cases, Jakarta EE 9 feature short names are the same as the Jakarta EE 8.0 features.  For these features, the Jakarta EE 9 feature names simply have an updated version number.  Here is a complete table of features which were updated by changing just the feature's version number:`

Reply:
I suggest changing `Java / Jakarta EE` just to `Jakarta EE 8` since EE9 is just EE 8 with different package names.

Reply:
See my comment, above.  I'm on the fence as to whether to use "java / Jakarta EE" or "Jakarta EE 8".

Reply:
Again it makes sense to me since Jakarta EE 9 is just Jakarate EE 8 with new names.  More in the follow on review comments.